capstone-backend/app.js

```diff
@@ -137,56 +138,340 @@ app.post ('/profile/interests', async (req, res) => {
+          let relations = skills.relation('User1');
+          relations.add(currentUser);
+          respSkills = await skills.save()
+        }
+      }
+      let respCompany = null
+      if (uiCompany != ""){
+        console.log('uiCompany: ', uiCompany);
+        const query = new Parse.Query(Company);
+        query.equalTo("User1", currentUser);
+        query.equalTo("name", uiCompany.name);
+        const results = await query.find();
+        console.log('results: ', results);
+        if (results.length != 0){
+          company.set("name", uiCompany.name);
+          let relations = company.relation("User1");
+          relations.add(currentUser);
+          respCompany = await company.save()
         }
       }
-      res.send({skills: skills, userInfo: currentUser, message: "interests saved!", typeStatus: "success", infoInterests: infoInterests})
+      if (uiLanguage){
+        const query = new Parse.Query(Language);
+        query.equalTo("User1", currentUser);
+        query.equalTo("name", uiLanguage.name);
+        const results = await query.find();
+        if (!results[0]){
+          language.set("name", uiLanguage.name);
+          let relations = language.relation("User1");
+          relations.add(currentUser);
+          await language.save()
+        }
+      }
+      res.send({skills: respSkills, company: respCompany, language: language,  userInfo: currentUser, message: "interests saved!", typeStatus: "success", infoInterests: infoInterests})
     } else {
-      res.send({skills: skills, userInfo: null, message: "cannot find a user currently", typeStatus: "danger", infoInterests: infoInterests})
+      res.send({skills: null , userInfo: null, message: "cannot find a user currently", typeStatus: "danger", infoInterests: infoInterests})
     }
   }
   catch(error){
     res.send( {message: error.message, infoInterests: infoInterests, typeStatus: "danger"})
   }
 })
 
+app.post('/profile/interests/remove', async (req, res) => {
+  const removeInfo = req.body
+  Parse.User.enableUnsafeCurrentUser();
+  const currentUser = Parse.User.current();
+  try {
+    if (currentUser){
+      if (removeInfo.skills){
+        removeInterest("Skills", "name", removeInfo.skills.name, currentUser)
+      }
+      res.send({message : "success", removeInfo: removeInfo, result: result[0]})
+    }
+    else {
+      res.send({message : "no user found", typeStatus: "danger"})
+    }
+  }
+  catch (error){
+    res.send({message: error.message, typeStatus: "danger"})
+  }
+})
+
+app.post('/matches', async (req, res) => {
+  Parse.User.enableUnsafeCurrentUser();
+  const params = req.body.params;
+  const currentUser = Parse.User.current()
+
+  try {
+    if (currentUser){
+      if (params.liked){
+        updateMatch(params, currentUser)
+      }
+      else {
+        getMatches(currentUser);
+      }
+      res.send({message: "success match", typeStatus: "success"})
+    }
+    else {
```

Comment:
It becomes more readable if we use early return instead of IF ELSE structure.
```
if (!currentUser) {
  res.send({message: "no user found!", typeStatus: "danger"});
  return;
}

// rest of the code here
```